presto-google-sheets/src/main/java/com/facebook/presto/google/sheets/SheetsModule.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.google.sheets;
+
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.common.type.TypeManager;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.FromStringDeserializer;
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.Scopes;
+
+import javax.inject.Inject;
+
+import static com.facebook.airlift.configuration.ConfigBinder.configBinder;
+import static com.facebook.airlift.json.JsonBinder.jsonBinder;
+import static com.facebook.airlift.json.JsonCodec.listJsonCodec;
+import static com.facebook.airlift.json.JsonCodecBinder.jsonCodecBinder;
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static java.util.Objects.requireNonNull;
+
+public class SheetsModule
+        implements Module
+{
+    private final TypeManager typeManager;
+
+    public SheetsModule(TypeManager typeManager)
+    {
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+    }
+
+    @Override
+    public void configure(Binder binder)
+    {
+        binder.bind(TypeManager.class).toInstance(typeManager);
+        binder.bind(SheetsConnector.class).in(Scopes.SINGLETON);
+        binder.bind(SheetsMetadata.class).in(Scopes.SINGLETON);
+        binder.bind(SheetsClient.class).in(Scopes.SINGLETON);
+        binder.bind(SheetsSplitManager.class).in(Scopes.SINGLETON);
+        binder.bind(SheetsRecordSetProvider.class).in(Scopes.SINGLETON);
+
+        configBinder(binder).bindConfig(SheetsConfig.class);
+
+        jsonBinder(binder).addDeserializerBinding(Type.class).to(TypeDeserializer.class);
+        jsonCodecBinder(binder).bindMapJsonCodec(String.class, listJsonCodec(SheetsTable.class));
+    }
+
+    public static final class TypeDeserializer
+            extends FromStringDeserializer<Type>
+    {
+        private final TypeManager typeManager;
+
+        @Inject
+        public TypeDeserializer(TypeManager typeManager)
+        {
+            super(Type.class);
+            this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        }
+
+        @Override
+        protected Type _deserialize(String value, DeserializationContext context)
```

Comment:
nit: Not sure about the coding style here. I think the Presto codebase names variables only in lowerCamelCase, which don't start with underscores.

Reply:
Agree that this is not quite the style here. However this is a method inherited from FromStringDeserializer (from com.fasterxml.jackson package). So I'm afraid it needs to be like this..

Reply:
Ah, I see. Thanks for your clarification! Yeah, in that case, we don't need to change it~